types/rdfjs__dataset/Factory.d.ts

```diff
@@ -0,0 +1,7 @@
+import { BaseQuad, Quad } from 'rdf-js';
+import DatasetCore from './DatasetCore';
+
+export default class Factory {
```

Comment:
Similarly - can this now become
```ts
export default class Factory<Q extends RDF.BaseQuad = RDF.Quad> extends RDF.DatasetCoreFactory<Q> {
   static readonly exports: ['dataset'];
}
```

Reply:
Here it's actually similar, the matter of returning the `DatasetCore` class or interface. Since it does not offer any extract functionality (unlike `rdf-ext`), we should be fine to use the interface too

Reply:
Ah, no, we cannot do this. The generic argument is on the `dataset<Q>` method. This is not the same, and will break the line in test: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ab71df0686214de5c17a7da19822aa8aa0ee9903/types/rdfjs__dataset/rdfjs__dataset-tests.ts#L20